protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java

```diff
@@ -814,6 +815,24 @@ private long parseUnsignedInteger(String value) throws DecodingException {
         boolean isValid(char chr);
     }
 
+    public static class StringValidator implements CharacterValidator {
+        private final byte[] expectedStringAsBytes;
+        private int position = 0;
+
+        public StringValidator(String expectedString) {
+            this.expectedStringAsBytes = expectedString.getBytes(StandardCharsets.US_ASCII);
+        }
+
+        @Override
+        public boolean isValid(char chr) {
+            if (position >= expectedStringAsBytes.length) {
+                return false;
+            } else {
+                return ImapRequestLineReader.cap(chr) == expectedStringAsBytes[position++];
```

Comment:
so if we call enough time isValid for the same char, we will always get false.
I am not sure to understand this thing at all.

Reply:
Well, that's a stateful validator, temporal coupling is expected.

You ensure that the character sequence you validate follow a given string pattern.

You are stuck with stateful cgaracter validation unless you deeply refactor IMAP requests.

I just intend to factorize and test duplicated code here, not fundamentaly rework the logic...